Dockerfile

```diff
@@ -12,10 +12,22 @@ COPY Gemfile /Gemfile
 COPY Gemfile.lock /Gemfile.lock
 RUN bundle install
 
+RUN apk --no-cache add curl unzip jq ca-certificates wget
+RUN wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub
+RUN wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.28-r0/glibc-2.28-r0.apk
+RUN apk add glibc-2.28-r0.apk
```

Comment:
I think that you can replace this line by: `apk add libc6-compat`. See this dockerfile:

```
FROM alpine:latest

ENV SDM_DISABLE_UPDATE=true
ENV SDM_DOCKERIZED=true
ENV SDM_HOME=/home/sdm/.sdm/
ENV PATH=$PATH:/home/sdm

RUN apk --no-cache add curl libc6-compat

# This block can be replaced by a COPY/ADD command
# Download sdm for always using the latest client
RUN curl -J -O -L https://app.strongdm.com/releases/cli/linux
RUN unzip -x sdm*.zip
RUN rm sdm*.zip
RUN apk del curl

RUN adduser -D sdm
RUN mv sdm /home/sdm

COPY entrypoint.sh /entrypoint.sh
RUN chmod +x /entrypoint.sh

# Key! Avoid root access
USER sdm 

ENTRYPOINT ["/entrypoint.sh"]
```

Reply:
When using only `libc6-compat`, we get the following errors:

```
log-export-container_1  | Error relocating ./sdm: __vfprintf_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memcpy_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memset_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __fprintf_chk: symbol not found
log-export-container_1  | Creating Fluentd conf file
log-export-container_1  | Error relocating ./sdm: __vfprintf_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memcpy_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __memset_chk: symbol not found
log-export-container_1  | Error relocating ./sdm: __fprintf_chk: symbol not found
```

We tried other approaches like using `gcompat` and other `glib` libraries, but it continues to display some errors meaning that it's still missing something.

Trying to run the dockerfile that you shared above in my Mac M1 Big Sur, it gives the following errors:

Without using DOCKER_DEFAULT_PLATFORM:

```
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
```

Using DOCKER_DEFAULT_PLATFORM=linux/amd64:

```
Error relocating /home/sdm/sdm: __vfprintf_chk: symbol not found
Error relocating /home/sdm/sdm: __memcpy_chk: symbol not found
Error relocating /home/sdm/sdm: __memset_chk: symbol not found
Error relocating /home/sdm/sdm: __fprintf_chk: symbol not found
```

Reply:
Try building in M1 with:
```
docker buildx build --platform linux/amd64 -t log-export-container .
```

Reply:
It returns the same error:

```
Starting SDM
Error relocating ./sdm: __vfprintf_chk: symbol not found
Error relocating ./sdm: __memcpy_chk: symbol not found
Error relocating ./sdm: __memset_chk: symbol not found
Error relocating ./sdm: __fprintf_chk: symbol not found
```

Reply:
It returns the same error:

```
Starting SDM
Error relocating ./sdm: __vfprintf_chk: symbol not found
Error relocating ./sdm: __memcpy_chk: symbol not found
Error relocating ./sdm: __memset_chk: symbol not found
Error relocating ./sdm: __fprintf_chk: symbol not found
```

Reply:
Perfect, merging then. Thank you guys